### ISSUE TITLE
Previews in Safari Version 15.1 MacOS do not authenticate

### ISSUE THREAD

**Issue by tbrasington** [sanity-io/next-sanity#35]
I am unable to get Safari to display previews, Chrome renders them fine. 

the usePreviewSubscription hook returns this error object within it's response

```
Error: Not authenticated - preview not available
column: 22
line: 180
message: "Not authenticated - preview not available"
stack: "@↵promiseReactionJob@[native code]"
```

My code
```typescript
  const data = usePreviewSubscription(getGuide, {
    params: { slug: slug },
    initialData: content,
    enabled: preview
  })
```

Also this console.warn message is shown in the console "Not authenticated - preview not available" within the function below from the next-sanity.esm.js file

```javascript
useDeepCompareEffectNoCheck(function () {
    if (!enabled) {
      return function () {
        /* intentional noop */
      };
    }

    setLoading(true);
    var aborter = getAborter();
    var subscription;
    getCurrentUser(projectId, aborter).then(function (user) {
      if (user) {
        return;
      } // eslint-disable-next-line no-console


      console.warn('Not authenticated - preview not available');
      throw new Error('Not authenticated - preview not available');
    }).then(getStore).then(function (store) {
      subscription = store.subscribe(query, params, function (err, result) {
        if (err) {
          setError(err);
        } else {
          setData(result);
        }
      });
    }).catch(setError).finally(function () {
      return setLoading(false);
    });
    return function () {
      if (subscription) {
        subscription.unsubscribe();
      }

      aborter.abort();
    };
  }, [getStore, query, params, enabled]);
  return {
    data: typeof data === 'undefined' ? initialData : data,
    loading: loading,
    error: error
  };
}
```

**Versions**
Safari is Version 15.1 (16612.2.9.1.30, 16612)

next app
```json
"@sanity/client": "^2.21.7",
"next": "^12.0.1",
"next-sanity": "^0.4.0", 
```

studio app
```json
"@sanity/base": "^2.21.9",
"@sanity/components": "^2.14.0",
"@sanity/core": "^2.21.8",
"@sanity/default-layout": "^2.21.9",
"@sanity/default-login": "^2.21.9",
"@sanity/desk-tool": "^2.21.9",
"@sanity/production-preview": "^2.15.0",
"@sanity/vision": "^2.21.9",
```

**Comment by Glavin001** [comment]
Same here. @tbrasington have you found a workaround?

**Comment by tbrasington** [comment]
> Same here. @tbrasington have you found a workaround?

Not for safari.

**Comment by stipsan** [comment]
Yes it's related to the cookie auth problem. [We're working on a new version](https://github.com/sanity-io/next-sanity/pull/65) of the preview tooling that supports the token flow, enabling Safari to use previews once more :)

**Comment by stipsan** [comment]
We've just released [`v2.0.0`](https://github.com/sanity-io/next-sanity/releases/tag/v2.0.0) which has docs that shows how you can use custom token auth that solves the problem you're experiencing here 😌
